test/src/test/java/hudson/PluginManagerTest.java

```diff
@@ -487,7 +487,7 @@ private String callDependerValue() throws Exception {
         // Check that the basic API endpoint invocation works.
         assertEquals("ok", response.getString("status"));
         JSONArray data = response.getJSONArray("data");
-        assertTrue(data.size() > 0);
+        assertFalse(data.isEmpty());
```

Comment:
both of these are not great - if it fails it does not give you hints as to why it fails - what was in the data is a great hint for debugging.

E.g. 
`assertThat(data, isEmpty())`  would be preferable here as it would produce something like `an empty collection but got ["some", "Collection", "with", "values"]`

Reply:
y'all should have bashed me over the head.  
as this would only fail if the collection was empty ther eis no additional information to be gleaned in this case.  an empty colleciton is an empty collection.  
The inverse (when checking it is not empty) is not the case - but these where asserting something was not empty, not that it *was* empty.

My appologies.

Reply:
I guess we all got a little confused :) But now it is really better than before :)